### PR TITLE
WIP: Isolate function calls

### PR DESCRIPTION
**Pull request by mlautman**
Residual changes in cwd as a byproduct of a function call without documentation is a bug not a feature. this uses pushd and popd to return to the initial cwd after each function call unless otherwise stated.

### CODE REVIEW

File: travis.sh [hunk 1]
```diff
@@ -31,6 +31,7 @@ function run_script() {
 }
 
 function run_docker() {
+  pushd .
```
Comment (rhaschke) [inline travis.sh]: Why do you introduce these noops?

File: travis.sh [hunk 2]
```diff
@@ -221,21 +225,24 @@ function prepare_ros_workspace() {
...
   travis_run --title "List files in ROS workspace's source folder" ls --color=auto -alhF
 
   # Install source-based package dependencies
   travis_run --retry rosdep install -y -q -n --from-paths . --ignore-src --rosdistro $ROS_DISTRO
 
-  # Change to base of workspace
-  travis_run_simple cd $ROS_WS
-
   # Validate that we have some packages to build
   test -z "$(catkin list)" && echo -e "$(colorize RED Workspace $ROS_WS has no packages to build. Terminating.)" && exit 1
```
Comment (rhaschke) [inline travis.sh]: There are exits and returns in many functions, which render your pushd/popd approach error-prone!
When returning from a function you would need to popd as well.

**Comment by mlautman** [comment]
Closing for now

**Comment by mlautman** [comment]
> I don't agree to these unnecessary changes.
> 
> 1. You know that I prefer tabs over spaces when it comes to indentation, because tabs allow the user to adjust the indentation depth to his/her preferences w/o changing the document. If you really insist in spaces, the changes shouldn't break Travis...

In the case of public projects, individual preferences should be weighed but style should be consistent. For the MoveIt project we use spaces instead of tabs and it would make sense to do the same here. At very minimum, a project should not mix tabs and spaces. I agree this change should not break travis and I am quite confused how this trivial change managed to break CI. I marked the PR WIP until it passes travis. 

> 2. The pushd/popd scheme only works if there are no other exit points from functions. Doing `pushd .` is useless and just clutters the code.

We (picknik) would like to use moveit_ci for a number of client projects rather than the private fork so that we can push back features such as supporting private repos, etc.. As it stands, the documentation for running moveit_ci locally is lacking making it hard for someone who isn't intimately familiar with the code to make changes. For maintainability, I think it is important that these methods are able to be run on their own. Using pushd and popd was intended to remove secondary effects of the methods. I would be open to adding a `cd` command to the beginning of each method as an alternative.  

> 3. Both changes introduces issues in Travis.

Working on it :)

**Comment by davetcoleman** [comment]
Per ROS and Google style guides:

> Indent each block by 2 spaces. Never insert literal tab characters.

It does seem @mlautman 's ``pushd`` changes are error-prone. Overall, this bash codebase is very hard to read and maintain, I'm a bit worried about that.

I'm re-opening this PR so it doesn't get lost, and I'd like to see most of these changes merged in for the reasons @mlautman listed.

**Pull request closed without merging**